knotx-core/knotx-knot-action/src/main/java/com/cognifide/knotx/knot/action/ActionKnotVerticle.java

```diff
@@ -134,12 +133,18 @@ private void handleFormAction(KnotContext knotContext, Handler<JsonObject> handl
           ClientResponse clientResponse = new ClientResponse(msg.body().getJsonObject("clientResponse"));
           String signal = msg.body().getString("signal");
 
+          if (!HttpResponseStatus.OK.equals(clientResponse.statusCode())) {
+            knotContext.clientResponse().setStatusCode(clientResponse.statusCode());
```

Comment:
O think, you must also copy headers from clientResponse to response in knotContext.